juspay/hyperswitch
# fix(connector): [Adyen] update config and add required fields

## Type of Change



## Description


- Update Prod URL endpoint
- Handle refunds webhooks using hyperswitch internal refund id
- Add card holder name

### Additional Changes



Updated Prod URL endpoint

## Motivation and Context

#4047 

## How did you test it?

- Trigger a refunds transaction
```
curl --location 'http://localhost:8080/refunds' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_gwfA0IT4jpKdcUSmJlWMJ0jawjc5ziP6lA91du0gpc9ydEzI5HajDN18Wr72NkS7' \
--data '{
    "payment_id": "pay_ow2YSNvgVDZlGTv3o8wc",
    
    "reason": "RETURN",
    "refund_type": "instant",
    "metadata": {
        "udf1": "value1",
        "new_customer": "true",
        "login_date": "2019-09-10T10:11:12Z"
    }
}'
```
- Check for outgoing webhook and also retrieve the refund to confirm the Status change
![Screenshot 2024-03-12 at 6 31 39 PM](https://github.com/juspay/hyperswitch/assets/55536657/b6195d05-7972-4079-b1f6-bdeee1fb1f2f)


- Send card holder name in the cards request it should be visible in Adyen Dashboard for the payment
```
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_gwfA0IT4jpKdcUSmJlWMJ0jawjc5ziP6lA91du0gpc9ydEzI5HajDN18Wr72NkS7' \
--data-raw '{
    "amount": 6540,
    "currency": "EUR",
    "confirm": true,
    "capture_method": "automatic",
    "capture_on": "2022-09-10T10:11:12Z",
    "amount_to_capture": 6540,
    "customer_id": "StripeCustomer",
    "email": "guest@example.com",
    "name": "John Doe",
    "phone": "999999999",
    "phone_country_code": "+1",
    "description": "Its my first payment request",
    "authentication_type": "no_three_ds",
    "return_url": "https://duck.com",
    "profile_id": "pro_8Ix85cVgHD3x4DQIFZTV",
    "business_country": "US",
    "payment_method": "card",
    "payment_method_type": "credit",
    "payment_method_data": {
        "card": {
            "card_number": "371449635398431",
            "card_exp_month": "03",
            "card_exp_year": "2030",
            "card_holder_name": "joseph Doe",
            "card_cvc": "7373"
        }
    },
    "routing": {
        "type": "single",
        "data": "adyen"
    },
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "DE",
            "first_name": "PiX"
        }
    },
    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "PiX"
        }
    },
    "statement_descriptor_name": "This is me making a statement",
    "statement_descriptor_suffix": "JS",
    "metadata": {
        "udf1": "value1",
        "new_customer": "true",
        "login_date": "2019-09-10T10:11:12Z"
    }
}'
```
## Checklist
